### ISSUE TITLE
Unexpected color output compared to Cmder/ConEmu

### ISSUE THREAD

**Issue by kojoru** [microsoft/terminal#2638]
# Environment

```none
Windows build number:  10.0.18362.0
Windows Terminal version : 0.4.2382.0
ConEmu version: 180626 
```

# Steps to reproduce
1. Copy the color palette for "<Solarized (John Doe)>" from ConEmu to terminal. Here's the relevant JSON:
```json
        {   
            "white" : "#fdf6e3",
            "blue" : "#073642",
            "green" : "#586e75",
            "cyan" : "#657b83",
            "red" : "#dc322f",
            "purple" : "#6c71c4",
            "yellow" : "#cb4b16",
            "brightWhite" : "#eee8d5",
            "brightBlack" : "#93a1a1",
            "brightBlue" : "#268bd2",
            "brightGreen" : "#859900",
            "brightCyan" : "#2aa198",
            "brightRed" : "#839496",
            "brightPurple" : "#d33682",
            "brightYellow" : "#b58900",
            "black" : "#002B36",
            "background" : "#002B36",
            "foreground" : "#fdf6e3",
            "name" : "Solarized (John Doe)"
        }
```
2. Set powershell's color scheme to "Solarized (John Doe)"
3. Run ` [enum]::GetValues([System.ConsoleColor]) | Foreach-Object {Write-Host $_ -ForegroundColor $_} ` to admire the colors

# Expected behavior

Colors look as nice as ConEmu's

# Actual behavior

Most colors look nice and similar but "DarkBlue" looks completely different and way worse. See screenshot. 
![image](https://user-images.githubusercontent.com/1836172/64105642-e860be00-cd76-11e9-8ed3-c79598061e66.png)

To get a somewhat similar color you need to set blue (DarkBlue) to something like `"blue" : "#0B5669"`
![image](https://user-images.githubusercontent.com/1836172/64105711-09c1aa00-cd77-11e9-8b39-1b5525650f92.png)

**Comment by j4james** [comment]
It looks to me like it's cmder that might be getting the color wrong. I suggest you try out those color values in an HTML document with a little test like this:

```
<body style="background:#002b36">
<div style="color:#073642">DarkBlue</div>
<div style="color:#586e75">DarkGreen</div>
<div style="color:#657b83">DarkCyan</div>
<div style="color:#dc322f">DarkRed</div>
<div style="color:#6c71c4">DarkMagenta</div>
</body>
```

As far as I can tell, that produces colors that exactly match what the Windows Terminal is showing.

**Comment by DHowett-MSFT** [comment]
It looks like ConEmu does some [cool stuff](https://github.com/Maximus5/ConEmu/blob/44fb97de59e663d9e81fafa31d95b0fe18c5a2fb/src/ConEmu/ColorFix.cpp#L488) to make sure that a color can always be _perceived_. It bumps colors that could be invisible on a certain background up in perceptibility using the [deltaE 2000](http://zschuessler.github.io/DeltaE/learn/) algorithm.

**Comment by DHowett-MSFT** [comment]
It looks like ConEmu lets you configure that with its "Adjust lightness of indistinguishable text" option. I'm betting that if you turn it off, you'll see the same colors as in WT.

**Comment by kojoru** [comment]
@DHowett fair enough! Should I then open a new issue to feature request the same in terminal or can we convert this one?

**Comment by WSLUser** [comment]
It would be pretty cool if we could do the same for Terminal. I had wondered why color output looked a bit different. Thanks for breaking it down @DHowett .

**Comment by hectormz** [comment]
What would be needed/considered to implement this in Terminal?

**Comment by DHowett** [comment]
All it would need is someone to do the work, and two people to review the pull request. :smile:

I'd prefer to see a specification to answer a couple questions: would this happen during rendering? Color scheme loading? Is there somewhere we could, should cache colors? How do we hook this setting up into the layer it needs to go into? Should we nudge rgb or 256-color entries the same way as user-specified ones?

**Comment by hectormz** [comment]
Three very easy things 😅

I can't answer any of the specifics regarding how this or WT actually work.

This would only pertain to foreground/background compatibility right?

But I would happy be with just user-specified colors. 

Nudging 256 color entries would be nice if it wasn't much more overhead, and was cached ahead of time. But looking beyond the scope of user-specified colors might cause this fix to grow too much. 

For example, opacity or background images would mess up these adjustments, so I don't think that should be supported/fixed.

**Comment by kojoru** [comment]
@CIPop I've been quite successfully using my patched version of `Solarized (John Doe)` snatched from ConEmu ever since I filed this issue. It goes as follows:
```
        {   
            "white" : "#fdf6e3",
            "blue" : "#0B5669",
            "green" : "#586e75",
            "cyan" : "#657b83",
            "red" : "#dc322f",
            "purple" : "#6c71c4",
            "yellow" : "#cb4b16",
            "brightWhite" : "#eee8d5",
            "brightBlack" : "#93a1a1",
            "brightBlue" : "#268bd2",
            "brightGreen" : "#859900",
            "brightCyan" : "#2aa198",
            "brightRed" : "#839496",
            "brightPurple" : "#d33682",
            "brightYellow" : "#b58900",
            "black" : "#002B36",
            "background" : "#002B36",
            "foreground" : "#fdf6e3",
            "name" : "Solarized (John Doe)"
        }
```

As compared to the built in Solarized, it never produces any unreadable results and has decent difference between colors.

**Comment by WSLUser** [comment]
@kojoru I suggest sending a PR updating the default with this instead.

**Comment by kojoru** [comment]
Fair enough! Will do.

**Comment by WSLUser** [comment]
This may help for anyone trying to implement this: http://www.easyrgb.com/en/math.php#text6. I got this website from the answer in https://stackoverflow.com/questions/5774152/calculate-how-humans-perceive-similarity-between-different-colours. We use chrome math functions in C++ so it just needs conversion to use that instead.
